layers/cmd/pipeline.go

```diff
@@ -136,6 +116,33 @@ func (fork ForkStep) Run(ctx *PipelineContext) error {
 	return nil
 }
 
+func forkPipeline(pipe Pipeline, ctx *PipelineContext, name string, errs map[string]error) {
+	pctx := &PipelineContext{
+		Context:     getContext(),
+		graph:       ctx.graph,
+		roots:       ctx.roots,
+		InputFiles:  ctx.InputFiles,
+		steps:       pipe,
+		currentStep: -1,
+		graphOwner:  ctx.graphOwner,
+	}
+	cpMap := make(map[string]interface{})
+	for k, prop := range ctx.Properties {
+		cpMap[k] = ls.ClonePropertyValueFunc(k, prop)
```

Comment:
No need for clonePropertyFunc here. Simply cpMap[k]=prop

Reply:
if there is no need for clonePropertyFunc. shouldnt i just assign pctx.Properties = ctx.Properties?